manpages/coeasy.1
.TH COIPS "7" "October 2012" "CompatibleOne" "Platform"
.SH NAME
coeasy \- CompatibleOne EasyClouds Service.
.SH SYNOPSIS
\fBcoeasy\fR
.PP
.SH DESCRIPTION
.\" Add any additional description here
.PP
This will run the CompatibleOne \fBcoeasy\fR service.
.SH AUTHOR
The CompatibleOne project.
.SH MANPAGE
Most of this documents comes from the README of ACCORDS, written by Iain James Marshall.
Written by Alexandre Lissy <alissy@mandriva.com> for the CompatibleOne project.
.SH "REPORTING BUGS"
Report CompatibleOne bugs to compatibleone@ow2.org
.br
CompatibleOne home page: <http://www.compatibleone.org>
.SH COPYRIGHT
Copyright \(co 2012 CompatibleOne.
License Apache 2.0.
.br
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.


debian/changelog
compatibleone-accords (1.0-6) unstable; urgency=low

  * Improved Python bindings.
  * Add 'compatibleone-cnprocci' package for ComputeNext interface.
  * Add 'compatibleone-eziprocci' package for EasyClouds interface.
  * Upstream code update.

 -- Vincent Renardias <vincent@renardias.com>  Mon, 11 Mar 2013 10:34:58 +0100

compatibleone-accords (1.0-5) unstable; urgency=low

  * Upstream update.
  * Fix dashboard scripts.

 -- Vincent <vincent@renardias.com>  Thu, 21 Feb 2013 22:17:00 +0100

compatibleone-accords (1.0-4) unstable; urgency=low

  * Add new package compatibleone-dashboard (more work needed).
  * Clean-up control file, start adding the php dashboard
  * Replace hardwired emails.

 -- Vincent <vincent@renardias.com>  Thu, 21 Feb 2013 18:49:00 +0100

compatibleone-accords (1.0-3) unstable; urgency=low

  * Upstream update (tree SHA: a5c42c620e43d7a698c35e725cb7cf7fbba61181)
  * Add libcocspi0 and libcocspi-dev packages (+bump up solib deps).

 -- Vincent <vincent@renardias.com>  Wed, 13 Feb 2013 12:18:57 +0100

compatibleone-accords (1.0-2) unstable; urgency=low

  * Upstream code of jan 23rd.
  * Include selftest.csp cordscript.
  * Remove occurences of paprocci in current build.

 -- Vincent <vincent@renardias.com>  Wed, 23 Jan 2013 14:58:01 +0100

compatibleone-accords (1.0-1) unstable; urgency=low

  * Bump up version number!
  * Update solib deps accordingly.
  * Include code for the current repository.
  * Code / Comments clean-ups in debian/rules.
  * Disable python testsuite (doesn't work and most likely won't be maintained).
  * Wrote minimalist manpage for the cosacs client to shut up lintian.
  * Declare compatibleone-accords and compatibleone-accords-dev as metapackages.
  * Packages are now fully clean according to lintian (except the Python JDK).

 -- Vincent Renardias <vincent@renardias.com>  Fri, 18 Jan 2013 14:01:40 +0100

compatibleone-accords (0.20130101-1) unstable; urgency=low

  * Upstream code update + .csp sample file.
  * Set dpkg-source 1.0 format.
  * Update lib soname dep.

 -- Vincent Renardias <vincent@renardias.com>  Wed, 02 Jan 2013 10:24:36 +0100

compatibleone-accords (0.20121218-1) unstable; urgency=low

  * New upstream code + more doc + sync all binary packages.

 -- Vincent Renardias <vincent@renardias.com>  Tue, 18 Dec 2012 14:43:00 +0200

compatibleone-accords (0.20121214-1) unstable; urgency=low

  * Several bug fixes regarding cosacs.

 -- Vincent Renardias <vincent@renardias.com>  Fri, 14 Dec 2012 16:58:00 +0200

compatibleone-accords (0.20121213-2) unstable; urgency=low

  * Add 'cosacs' package (client side COSACS to be run in the VM images).
  * Split 'compatibleone' in 'compatibleone-base' and 'compatibleone'.

 -- Vincent Renardias <vincent@renardias.com>  Thu, 13 Dec 2012 17:47:00 +0200

compatibleone-accords (0.20121213-1) unstable; urgency=low

  * Major renaming of packages to be fully Debian compliant and prepare the
    integration of the Cosacs client.
  * Force lib # in the dependency and bump the version to make sure we're
    not linking against a version older than today (lib ABI change).

 -- Vincent Renardias <vincent@renardias.com>  Tue, 12 Dec 2012 15:16:00 +0200

compatibleone-accords (0.20121212-1) unstable; urgency=low

  * Upstream update.
  * Major packaging update: each lib now has its own packages.

 -- Vincent Renardias <vincent@renardias.com>  Tue, 11 Dec 2012 15:47:00 +0200

compatibleone-accords (0.20121207-1) unstable; urgency=low

  * Upstream update.
  * Lots of packaging fixes (see gitorious log for the gory details)

 -- Vincent Renardias <vincent@renardias.com>  Fri, 07 Dec 2012 10:25:21 +0200

compatibleone-accords (0.20121129-1) unstable; urgency=low

  * Upstream update.

 -- Vincent Renardias <vincent@renardias.com>  Mon, 26 Nov 2012 11:43:34 +0200

compatibleone-accords (0.20121121-1) unstable; urgency=low

  * Uptream update 2012/11/21 codebase + python enabled.
  * compatibleone-example and compatibleone-testosocci are now gone.

 -- Vincent Renardias <vincent@renardias.com>  Wed, 21 Nov 2012 11:12:59 +0200

compatibleone-accords (0.20121120-1) unstable; urgency=low

  * Uptream update 2012/11/20 codebase + perm fixes.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Tue, 20 Nov 2012 15:24:44 +0200

compatibleone-accords (0.20121114-2) unstable; urgency=low

  * Uptream update 2012/11/19 codebase.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Mon, 19 Nov 2012 20:41:06 +0200

compatibleone-accords (0.20121114-1) unstable; urgency=low

  * Uptream update,
  * jpaasprocci -> paasprocci renaming.
  * fixes for java build on 64bit arch.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Wed, 14 Nov 2012 14:24:57 +0200

compatibleone-accords (0.20121022-1) unstable; urgency=low

  * Upstream change: new accords bash script.
  * All executables now have a manpage.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Mon, 22 Oct 2012 15:18:00 +0200

compatibleone-accords (0.20121015-1) unstable; urgency=low

  * Only recommend compatibleone-paprocci so everybody isn't forced to install the Java runtime.
  * psmisc dep so that we're sure killall is installed.
  * Change OpenSSL dependancy so it also builds on Debian 6.0.x
  * Change debhelper dep so the package builds on Ubuntu 9.04.
  * Commit work with Mauricio to build compatibleone-paprocci and run it with
    the appropriate autodetected java environment.
  * co-stop: can now be run from any directory.
  * Add package compatibleone-example.
  * Add compatibleone-nagios package.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Mon, 15 Oct 2012 11:08:00 +0200

compatibleone-accords (0.20121008-1) unstable; urgency=low

  * More fixes/updates (see gitorious log).

 -- Vincent Renardias <vrenardias@citypassenger.com>  Mon, 8 Oct 2012 14:38:16 +0200

compatibleone-accords (0.20121004-1) unstable; urgency=low

  * More fixes/updates, Initial init script for publisher, WIP.


coeasy/src/coeasy.c
#ifndef	_coeasy_c	
#define	_coeasy_c

#include "standard.h"
#include "broker.h"
#include "rest.h"
#include "occi.h"
#include "document.h"
#include "cordspublic.h"
#include "occipublisher.h"

/* 	----------------------------------------	*/	
/*	Prototype Contract Negotiation Functions	*/
/* 	----------------------------------------	*/

struct	accords_configuration CoEasy = {
	0,0,
	0,0,0,0,
	(char *) 0,
	(char *) 0, 
	_CORDS_DEFAULT_PUBLISHER,
	_CORDS_DEFAULT_OPERATOR,
	_CORDS_DEFAULT_USER,
	_CORDS_DEFAULT_PASSWORD,
	"http",  80,
	"xmpp",  8000,
	"domain",
	"europe",
	"coeasy.xml",
	"storage",
	(struct occi_category *) 0,
	(struct occi_category *) 0
	};

public	int	check_debug()		{	return(CoEasy.debug);		}
public	int	check_verbose()		{	return(CoEasy.verbose);		}
public	char *	default_publisher()	{	return(CoEasy.publisher);	}
public	char *	default_operator()	{	return(CoEasy.operator);		}
public	char *	default_tls()		{	return(CoEasy.tls);		}
public	char *	default_zone()		{	return(CoEasy.zone);		}

public	int	failure( int e, char * m1, char * m2 )
{
	if ( e )
	{
		printf("\n*** failure %u",e);
		if ( m1 )
			printf(" : %s",m1);
		if ( m2 )
			printf(" : %s",m2);
		printf(" **`\n");
	}
	return( e );
}

#include "easiclouds_application.c"
#include "easiclouds_connection.c"
#include "easiclouds_description.c"
#include "easiclouds_metadata.c"
#include "easiclouds_extra.c"
#include "easiclouds_instance.c"
#include "easiclouds_keypair.c"
#include "easiclouds_link.c"
#include "easiclouds_model.c"
#include "easiclouds_node.c"
#include "easiclouds_parameters.c"
#include "easiclouds_postconf.c"
#include "easiclouds_server.c"

#include "occieasiclouds_application.c"
#include "occieasiclouds_connection.c"
#include "occieasiclouds_description.c"
#include "occieasiclouds_metadata.c"
#include "occieasiclouds_extra.c"
#include "occieasiclouds_instance.c"
#include "occieasiclouds_keypair.c"
#include "occieasiclouds_link.c"
#include "occieasiclouds_model.c"
#include "occieasiclouds_node.c"
#include "occieasiclouds_parameters.c"
#include "occieasiclouds_postconf.c"
#include "occieasiclouds_server.c"


/*	---------------------------------------------------------------	*/  
/*	c o e a s y _ c o n f i g u r a t i o n				*/
/*	---------------------------------------------------------------	*/  
/*	this function loads coeasy configuration			*/
/*	from the xml configuration file.				*/
/*	---------------------------------------------------------------	*/  
private	void	coeasy_configuration()
{
	load_accords_configuration( &CoEasy, "coeasy" );
	return;
}

/*	-----------------------------------------------------	----------	*/  
/*	c o e a s y _ b a n n e r				*/
/*	---------------------------------------------------------------	*/  
/*	this function will be called to display module identification 	*/
/*	when launched from the command line without any parameters.	*/
/*	---------------------------------------------------------------	*/  
private	int	coeasy_banner()
{
	printf("\n   CompatibleOne Easiclouds CoEasy Service : Version 1.0a.0.01");
	printf("\n   Beta Version : 06/03/2013 ");
	printf("\n   Copyright (c) 2013 Iain James Marshall ");
	printf("\n");
	accords_configuration_options();
	printf("\n\n");
	return(0);

}

/*	------------------------------------------- */
/*	c o e a s y _ i n i t i a l i s e	    */
/*	------------------------------------------- */

private	struct rest_server * coeasy_initialise(  void * v,struct rest_server * sptr )
{
	struct	rest_extension * xptr;
	if (!( xptr = rest_add_extension( sptr ) ))
		return((struct rest_server *) 0);
	else
	{
		xptr->net = (struct connection *) 0;
		return( sptr );
	}
}

/*	-------------------------------------------	*/
/*	c o e a s y _ a u t h o r i s e 		*/
/*	------------------------------------------- 	*/
private	int	coeasy_authorise(  void * v,struct rest_client * cptr, char * username, char * password )
{
	if ( strcmp( username, CoEasy.user ) )
		return(0);
	else if ( strcmp( password, CoEasy.password ) )
		return(0);
	else if (!( cptr->user = allocate_string( username ) ))
		return(0);
	else if (!( cptr->pass = allocate_string( password ) ))
		return(0);
	else	return(1);
}

/*	-------------------------------------------	*/
/*	c o e a s y _ e x t e n s i o n 		*/
/*	-------------------------------------------	*/
private	struct rest_extension * coeasy_extension( void * v,struct rest_server * sptr, struct rest_extension * xptr)
{
	return( xptr );
}

/*	--------------------------------------------	*/
/*	c o e a s y _ o p e r a t i o n 		*/
/*	--------------------------------------------	*/
/*	environment and category preparation		*/
/*	--------------------------------------------	*/
private	int	coeasy_operation( char * nptr )
{

	struct	occi_category * first=(struct occi_category *) 0;
	struct	occi_category * last=(struct occi_category *) 0;
	struct	occi_category * optr=(struct occi_category *) 0;

	set_autosave_cords_xlink_name("links_coeasy.xml");

	rest_initialise_log( CoEasy.monitor );

	/* ---------------------------------------- */
	/* preparation of application category list */
	/* ---------------------------------------- */

	if (!( optr = occi_easiclouds_application_builder( CoEasy.domain, "easiclouds_application" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_connection_builder( CoEasy.domain, "easiclouds_connection" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_description_builder( CoEasy.domain, "easiclouds_description" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_extra_builder( CoEasy.domain, "easiclouds_extra" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_instance_builder( CoEasy.domain, "easiclouds_instance" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_keypair_builder( CoEasy.domain, "easiclouds_keypair" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_link_builder( CoEasy.domain, "easiclouds_link" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_model_builder( CoEasy.domain, "easiclouds_model" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_node_builder( CoEasy.domain, "easiclouds_node" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_parameters_builder( CoEasy.domain, "easiclouds_parameters" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_postconf_builder( CoEasy.domain, "easiclouds_postconf" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_metadata_builder( CoEasy.domain, "easiclouds_metadata" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( optr = occi_easiclouds_server_builder( CoEasy.domain, "easiclouds_server" ) ))
		return( 27 );
	else if (!( optr->previous = last ))
		first = optr;
	else	optr->previous->next = optr;
	last = optr;

	if (!( CoEasy.identity ))
	{
		/* --------------------------	*/
		/* launch simple occi server 	*/
		/* --------------------------	*/
		return( occi_server(  
			nptr, 
			CoEasy.restport, CoEasy.tls, 
			CoEasy.threads, first, (char *) 0 ) );
	}
	else
	{
		/* --------------------------------------------------------------- */
		/* publication and pricing of categories before occi server launch */
		/* --------------------------------------------------------------- */
		initialise_occi_publisher( 
			CoEasy.publisher, (char*) 0, (char *) 0, (char *) 0);

		return( publishing_occi_server(
				CoEasy.user, CoEasy.password,
				CoEasy.identity,  nptr, 
				CoEasy.restport, CoEasy.tls, 
				CoEasy.threads, first ) );
	}
}

/*	------------------------------------------- */
/*	c o e a s y _ o p t i o n s		    */
/*	------------------------------------------- */
/*	Command line option analysis		    */
/*	------------------------------------------- */
private	int	coeasy_options(int argc, char * argv[] )
{
	int	status=0;
	int	argi=0;
	char *	aptr;
	coeasy_configuration();
	while ( argi < argc )
	{
		if (!( aptr = argv[++argi] ))
			break;
		else if ( *aptr == '-' )
		{
			aptr++;
			switch( *(aptr++) )
			{
			case	'v'	:
				CoEasy.verbose=1;
				continue;
			case	'd'	:
				CoEasy.debug = 0xFFFF;
				continue;
			case	'-'	:
				if (!( argi = accords_configuration_option( aptr, argi, argv )))
					break;
				else	continue;
			}
			status = 30;
			break;
		}
		else if (!( status = coeasy_operation(aptr) ))
			continue;
		else	break;
	}
	return(status);
}

/*	--------------------------------------------	*/
/*	m a i n 					*/
/*	--------------------------------------------	*/
/*	Entry point from the command line		*/
/*	--------------------------------------------	*/
public	int	main(int argc, char * argv[] )
{
	if ( argc == 1 )
		return( coeasy_banner() );
	else	return( coeasy_options( argc, argv ) );
}

	/* ---------- */
#endif 	/* _coeasy_c */
	/* ---------- */


